### DISCUSSION TITLE
Support OpenID Connect Pushed Authorization Request (PAR) and JWT Authorization Request (JAR)

### DISCUSSION THREAD

**Discussion by illoyd** [directus/directus#20300] [Feature Requests]
### Summary

Please support the OpenID Connect PAR and JAR formats.

PAR and JAR are supported by `openid-client`.

https://github.com/panva/node-openid-client/blob/main/docs/README.md#clientpushedauthorizationrequestparameters-extras

An additional implication of moving to this security is supporting authorization codes returned as fragments (using the hash symbol in the URL) instead of queries (using the question mark symbol in the URL).

The callback endpoint needs to handle not receiving the auth code as a query, meaning the server will not receive the auth code. The browser needs to detect it has received the auth code in the fragment and then submit the auth code to the server either by redirecting to use the fragment as a query string, or by doing a direct POST to the callback endpoint.

### Motivation

To be compliant with advanced security considerations of OpenID Connect.

### Detailed Design

* To support the PAR and JAR flows either inside the OpenIDAuthDriver or as a new OpenIDConnectAuthDriver
* To 

### Must Have:
- Support PAR
- Support auth codes as fragments, not queries

### Could Have:
- D Support JAR

### Drawbacks

Directus is not compliant with the future direction of OpenID.

### Alternatives

Could be built as a new AuthDriver extension, if allowed.

### Adoption Strategy

New AuthDriver.

### Unresolved Questions

OpenID Connect FAPI 1 and FAPI 2 are complex; in an ideal world, we could autoconfigure the `openid-client` `Client` by the `Issuer` to handle all this.

However, given it is complex I would recommend putting it in its own OpenIDConnectAuthDriver, separate from the more basic OpenIDAuthDriver.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
